src/zero_copy/primitive.rs

```diff
@@ -336,19 +333,90 @@ pub const fn any<I: Input + ?Sized, E: Error<I>, S>() -> Any<I, E, S> {
     }
 }
 
-pub struct TakeUntil<P, I: ?Sized, C = (), E = (), S = ()> {
-    until: P,
-    phantom: PhantomData<(C, E, S, I)>,
+/// See [`filter_map`].
+#[must_use]
+pub struct FilterMap<'a, F, I: ?Sized, O, E = (), S = ()>(F, PhantomData<(&'a I, O, E, S)>);
```

Comment:
I believe that the intention was to replace this with `any().try_map(...)`, or at least have the `filter_map` function just be a light wrapper over it.

Reply:
Ah, thanks. I've added it because I used it in my project and I didn't know about try_map().